linq4j/src/main/java/org/apache/calcite/linq4j/function/Functions.java

```diff
@@ -623,10 +620,37 @@ public static int compareLists(List<?> b0, List<?> b1) {
     return Integer.compare(b0.size(), b1.size());
   }
 
+  /**
+   * Compares two maps.
+   *
+   * <p>Since maps in Calcite are implemented using {@link java.util.LinkedHashMap},
```

Comment:
not crazy about this. Is the order of insertion of elements in a map deterministic? Probably not in general, e.g., if you use an aggregate to build the map. 

In our implementation the maps are actually sorted by key. What do other system say about this?

Reply:
My understanding is that a LinkedHashMap can maintain the insertion order, allowing for comparisons based on that sequence. Initially, I sorted the data before comparing, but I found that this approach was inconsistent with DuckDB's behavior. The main issue is that I couldn’t find another database to compare results with. Honestly, I don’t have a strong preference for whether sorting should be done before comparison.

Reply:
yes, linkedhashmap retains insertion order, but that order is not defined in general for SQL, e.g., `SELECT MAP(SELECT a, b FROM x)` does not guarantee a specific row order.

You can leave it as it is, but the results are non-deterministic as defined.